includes/class-nrh.php

```diff
@@ -17,99 +17,451 @@ class NRH {
+		$gtag_amp_opt['vars']['config'][ $ga_property_code ]['linker']['domains'][]      = 'checkout.fundjournalism.org';
+		$gtag_amp_opt['vars']['config'][ $ga_property_code ]['linker']['decorate_forms'] = true;
+		return $gtag_amp_opt;
 	}
 
 	/**
-	 * Add the NRH checkout URL as an allowed redirect target.
-	 *
-	 * @param array $hosts Array of allowed hosts.
-	 * @return array Modified $hosts.
+	 * Get the site URL with protocol removed.
 	 */
-	public static function allow_redirect_to_nrh( $hosts ) {
-		$hosts[] = 'checkout.fundjournalism.org';
-		return $hosts;
+	public static function get_clean_site_url() {
+		$protocols = [ 'http://', 'https://' ];
+		return str_replace( $protocols, '', get_bloginfo( 'wpurl' ) );
 	}
 
 	/**
-	 * Add a hidden campaign input when a custom campaign is present in the GET request.
+	 * Rewrite Donate block for News Revenue Hub submission.
 	 *
 	 * @param string $html The donate form html.
+	 * @param array  $attributes Block attributes.
+	 *
 	 * @return string modified $html.
 	 */
-	public static function handle_custom_campaign_id( $html ) {
-		// Don't add a global campaign ID if there is already a campaign ID.
-		if ( stripos( $html, "name='campaign'" ) || stripos( $html, 'name="campaign"' ) ) {
-			return $html;
-		}
+	public static function render_nrh_donate_block( $html, $attributes ) {
+			$settings = array_merge( Donations::get_donation_settings(), $attributes );
 
-		$custom_campaign = filter_input( INPUT_GET, 'campaign', FILTER_SANITIZE_STRING );
-		if ( ! $custom_campaign ) {
-			return $html;
-		}
+			$frequencies = [
+				'once'  => __( 'One-time', 'newspack-blocks' ),
+				'month' => __( 'Monthly', 'newspack-blocks' ),
+				'year'  => __( 'Annually', 'newspack-blocks' ),
+			];
 
-		$custom_campaign = '<input type="hidden" name="campaign" value="' . esc_attr( $custom_campaign ) . '"/>';
-		$html            = str_replace( '</form>', $custom_campaign . '</form>', $html );
-		return $html;
+			$nrh_frequency_keys = [
+				'once'  => 'once',
+				'month' => 'monthly',
+				'year'  => 'yearly',
+
+			];
+
+			$selected_frequency = isset( $attributes['defaultFrequency'] ) ? $attributes['defaultFrequency'] : 'month';
+			$suggested_amounts  = $settings['suggestedAmounts'];
+
+			$campaign = isset( $attributes['campaign'] ) ? $attributes['campaign'] : false;
+
+			$uid = wp_rand( 10000, 99999 ); // Unique identifier to prevent labels colliding with other instances of Donate block.
+
+			$button_text = $attributes['buttonText'];
+
+			$nrh_config = get_option( NEWSPACK_NRH_CONFIG );
+
+			$organization_id = wp_strip_all_tags( $nrh_config['nrh_organization_id'] );
+
+			$allowed_tags = [
+				'form'  => [
+					'method' => [],
+					'action' => [],
+				],
+				'input' => [
+					'type'  => [],
+					'name'  => [],
+					'value' => [],
+				],
+			];
+
+			$is_amp        = function_exists( 'is_amp_endpoint' ) && is_amp_endpoint();
+			$block_id      = 'wpbdbd_' . uniqid();
```

Comment:
Not a blocker or anything, just curious what does `wpbdbd` stand for? WordPress Block Donate Block Donate? :)

Reply:
Exactly - this isn't new in this branch, but at some point @laurelfulford and I realized we were adding a substantial amount of CSS weight by using verbose block names, which was pushing sites right up to the AMP CSS limit. Here's one PR where this change was implemented: https://github.com/Automattic/newspack-blocks/pull/184